CMakeLists.txt

```diff
@@ -216,8 +217,10 @@ endif()
 
 add_library(sparrow SHARED ${SPARROW_HEADERS} ${SPARROW_SRC})
 
+set_target_properties(sparrow PROPERTIES VERSION ${CMAKE_PROJECT_VERSION}
+                              SOVERSION ${PROJECT_VERSION_MAJOR})
```

Comment:
We need to introduce binary versionning (which is different from API versionning) and set the so version accordingly.